tools/api-coverage/src/main.rs

```diff
@@ -0,0 +1,207 @@
+        );
+
+        println!(
+            "Called APIs:   {} of {} ({:.2}%)",
+            called_apis,
+            generated_apis,
+            called_apis as f64 / generated_apis as f64 * 100.0f64
+        );
+
+        mismatch_constants || mismatch_apis
+    }
+
+    fn report_constants(&self) -> (bool, u64, u64) {
+        let mut constant_names: Vec<String> = Vec::new();
+        for key in self.generated_defs.constants.keys() {
+            constant_names.push(key.clone());
+        }
+        constant_names.sort();
+
+        let mut unwrapped: Vec<String> = Vec::new();
+        let mut declared = 0;
+        let mut generated = 0;
+        let mut mismatch = false;
+
+        for name in constant_names {
+            generated += 1;
+            let constant = self.generated_defs.constants.get(&name).unwrap();
+            if let Some(ignored) = self.ignored_defs.constants.get(&name) {
+                assert_eq!(constant.expr, ignored.expr);
+                continue;
+            }
+
+            let sys_def = self.sys_defs.constants.get(&name);
+            if sys_def.is_none() {
+                unwrapped.push(name);
+            } else if sys_def.unwrap().expr == constant.expr {
+                declared += 1;
+            } else {
+                mismatch = true;
+                println!("CONSTANT MISMATCH: {}", name);
+            }
+        }
+
+        if mismatch {
+            println!();
+        }
+
+        if !unwrapped.is_empty() {
+            println!("Unwrapped Constants:");
+            for name in &unwrapped[..] {
+                println!("  {}", name);
+            }
+            println!();
+        }
+
+        (mismatch, declared, generated)
+    }
+
+    fn report_calls(&self) -> (bool, u64, u64, u64) {
+        let mut signature_names: Vec<String> = Vec::new();
+        for key in self.generated_defs.signatures.keys() {
+            signature_names.push(key.clone());
+        }
+        signature_names.sort();
+
+        let mut uncalled: Vec<String> = Vec::new();
+        let mut unwrapped: Vec<String> = Vec::new();
+        let mut called = 0;
+        let mut declared = 0;
+        let mut generated = 0;
+        let mut mismatch = false;
+
+        for name in signature_names {
+            generated += 1;
+            let sig = self.generated_defs.signatures.get(&name).unwrap();
+            if let Some(ignored) = self.ignored_defs.signatures.get(&name) {
+                assert_eq!(sig.inputs, ignored.inputs);
+                continue;
+            }
+
+            let mut wrapped = false;
+            let sys_def = self.sys_defs.signatures.get(&name);
+            if sys_def.is_none() {
+                unwrapped.push(name.clone());
+            } else if sys_def.unwrap().inputs == sig.inputs {
+                wrapped = true;
+                declared += 1;
+            } else {
+                mismatch = true;
+                println!("SIGNATURE MISMATCH: {}", name);
```

Comment:
Should this, and the other mismatches, go to stderr?

Reply:
I don't think so. My main intention for going this far is to eventually have this be a required check in CI. The mismatch case is to specifically catch when a C API function's arguments change. As such, that means we'll want the report to be consistent which writing to separate streams can mess up.

Reply:
Also for the CI angle, I also intend to exit non-zero after we've reached 100% on everything and then suddenly we detect less than 100% as that'll automatically flag any changes to the C API for us going forward.